### PR TITLE
Added Azure ARM template for creating Device Provisioning Service in …

### PR DESCRIPTION
**Pull request by nasirvi**
…all local, basic and standard deployment

# Description and Motivation 
Code changes in Azure Template to deploy `[IoT Hub Device Provisioning Service](https://docs.microsoft.com/en-us/azure/iot-dps/)` as part of Remote Monitoring solution deployment. With this changes DPS will deployed in all local, basic and standard deployments.

I have tested changes by deploying the local and basic Remote Monitoring solution, verified that WebUI is working with telemetry and containers are running without any error. I have also auto-provisioned a simulated TPM device using the Azure IoT C SDK by following steps mentioned in https://docs.microsoft.com/en-us/azure/iot-dps/quick-create-simulated-device.

Issue Link: https://github.com/Azure/pcs-cli/issues/333
...

# Change type 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

**Checklist:**

- [ ] All tests passed
- [x] The code follows the code style and conventions of this project
- [x] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

---
This change is [Reviewable](https://reviewable.io/reviews/azure/pcs-cli/334)

### CODE REVIEW

File: solutions/remotemonitoring/armtemplates/basic-static-map.json [hunk 1]
```diff
@@ -653,6 +661,28 @@
...
+                "iotHubs": [
+                    {
+                        "connectionString": "[concat('HostName=', reference(variables('iotHubResourceId')).hostName, ';SharedAccessKeyName=', variables('iotHubKeyName'), ';SharedAccessKey=', listkeys(variables('iotHubKeyResource'), variables('iotHubApiVersion')).primaryKey)]",
+                        "location": "[variables('location')]"
+                    }
+                ]
+            },
+            "sku": {
+                "name": "S1",
+                "capacity": 1
```
Comment (ppathan) [inline solutions/remotemonitoring/armtemplates/basic-static-map.json]: Does this capacity needs to match the IotHub Skus? What if the parameter for sku is S2 or S3?

File: solutions/remotemonitoring/armtemplates/basic-static-map.json [hunk 2]
```diff
@@ -653,6 +661,28 @@
...
+            "properties": {
+                "iotHubs": [
+                    {
+                        "connectionString": "[concat('HostName=', reference(variables('iotHubResourceId')).hostName, ';SharedAccessKeyName=', variables('iotHubKeyName'), ';SharedAccessKey=', listkeys(variables('iotHubKeyResource'), variables('iotHubApiVersion')).primaryKey)]",
+                        "location": "[variables('location')]"
+                    }
+                ]
+            },
+            "sku": {
+                "name": "S1",
```
[the remark this answers is not in the document]
Comment (nasirvi) [reply solutions/remotemonitoring/armtemplates/basic-static-map.json]: Actually DPS Sku name is different from IoT Hub Sku name. Only Allowed value for DPS Sku name is "S1".

Reference:
- https://docs.microsoft.com/en-us/azure/templates/microsoft.devices/provisioningservices#iotdpsskuinfo-object
- https://azure.microsoft.com/en-us/pricing/details/iot-hub/

Should I add it as parameter and set appropriate "allowedValues"?

File: solutions/remotemonitoring/armtemplates/basic-static-map.json (hunk 1 above)
Comment (nasirvi) [reply solutions/remotemonitoring/armtemplates/basic-static-map.json]: This is different than capacity parameter of IotHub Skus and represents the number of DPS units to provision.

File: solutions/remotemonitoring/armtemplates/basic-static-map.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (ppathan) [reply solutions/remotemonitoring/armtemplates/basic-static-map.json]: Adding as allowedValues would be great. Thanks!

[the remark this answers is not in the document]
Comment (nasirvi) [reply solutions/remotemonitoring/armtemplates/basic-static-map.json]: Added parameter with allowedValues for device provisioning sku.

**Pull request merged**